src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -875,25 +876,18 @@ public ImmutableArray<T> RemoveAll(Predicate<T> match)
             Requires.NotNull(match, "match");
 
             if (self.IsEmpty)
-            {
-                return new ImmutableArray<T>(self.array);
-            }
+                return self;
 
-            List<int> removeIndexes = null;
+            List<int> removeIndexes = new List<int>();
```

Comment:
@sharwell Why not? The [`List<T>` ctor](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Collections/Generic/List.cs#L49-L51) is extremely cheap so there's no benefit to lazy initialization. We also avoid a null check for every match, and make the code more readable.